lib/models/ColdStorageActionQueueEntry.js

```diff
@@ -0,0 +1,88 @@
+const joi = require('joi');
+
+const entrySchema = joi.object({
+    op: joi.string().valid('restore', 'delete', 'archive').required(),
+    bucketName: joi.string().required(),
+    objectKey: joi.string().required(),
+    objectVersion: joi.string().required(),
+    accountId: joi.string().required(),
+    archiveInfo: joi.object().required(),
+    requestId: joi.string().required(),
+});
+
+class ColdStorageActionQueueEntry {
+    static createFromKafkaEntry(kafkaEntry) {
+        const res = entrySchema.validate(kafkaEntry.value);
+
+        if (res.error) {
+            return {
+                error: {
+                    message: 'malformed JSON in kafka entry',
+                    description: res.error,
+                },
+            };
+        }
+
+        const action = new ColdStorageActionQueueEntry(res.value);
+        return action;
+    }
+
+    constructor(attributes) {
+        this._attributes = attributes;
+    }
+
+    set(key, value) {
+        this._attributes = value;
+    }
+
+    get(key) {
+        return this._attributes[key];
+    }
+
+    getOp() {
+        return this._attributes.op;
+    }
+
+    getArchiveInfo() {
```

Comment:
I feel like we should use getters for this sort of stuff https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get